### ISSUE TITLE
Code not generated in Flutter

### ISSUE THREAD

**Issue by josephmangmang** [Jaguar-dart/jaguar_serializer#177]
Always failed. I'm using the latest version.
>>> NoSuchMethodError: The getter 'displayName' was called on null.

[INFO] Generating build script...
[INFO] Generating build script completed, took 682ms

[INFO] Initializing inputs
[INFO] Reading cached asset graph...
[INFO] Reading cached asset graph completed, took 194ms

[INFO] Checking for updates since last build...
[INFO] Checking for updates since last build completed, took 1.7s

[INFO] Running build...
[INFO] Running build completed, took 33ms

[INFO] Caching finalized dependency graph...
[INFO] Caching finalized dependency graph completed, took 95ms

[SEVERE] jaguar_serializer_cli:jaguar_serializer_cli on lib/src/models/hashtag.dart (cached):
Error running JaguarSerializerGenerator
NoSuchMethodError: The getter 'displayName' was called on null.
Receiver: null
Tried calling: displayName
[SEVERE] Failed after 154ms
pub failed (1)

**Comment by matthewrice345** [comment]
+1

**Comment by matthewrice345** [comment]
If I comment out my test dependencies I can successfully run the CLI
`  flutter_test:
    sdk: flutter
  test:`

**Comment by escamoteur** [comment]
Same here, I'm blocked because of that

**Comment by SamuelDebruyn** [comment]
Same here...

Verbose output:

```
Error running JaguarSerializerGenerator
NoSuchMethodError: The getter 'displayName' was called on null.
Receiver: null
Tried calling: displayName
dart:core                                                               Object.noSuchMethod
package:jaguar_serializer_cli/src/writer/writer.dart 129:43             Writer._writeCtor.<fn>
dart:core                                                               List.forEach
package:jaguar_serializer_cli/src/writer/writer.dart 126:24             Writer._writeCtor
package:jaguar_serializer_cli/src/writer/writer.dart 107:5              Writer._fromWriter
package:jaguar_serializer_cli/src/writer/writer.dart 31:5               Writer.generate
package:jaguar_serializer_cli/src/generator.dart 41:14                  JaguarSerializerGenerator.generateForAnnotatedElement
package:source_gen/src/generator_for_annotation.dart 47:28              GeneratorForAnnotation.generate
package:source_gen/src/builder.dart 254:35                              _generate
package:source_gen/src/builder.dart 72:15                               _Builder._generateForLibrary
package:source_gen/src/builder.dart 66:11                               _Builder.build
package:build                                                           runBuilder
package:build_runner_core/src/generate/build_impl.dart 460:19           _SingleBuild._runForInput.<fn>.<fn>.<fn>
package:build_runner_core/src/generate/performance_tracker.dart 303:15  _NoOpBuilderActionTracker.trackStage
package:build_runner_core/src/generate/build_impl.dart 458:23           _SingleBuild._runForInput.<fn>.<fn>
package:timing/src/timing.dart 222:44                                   NoOpTimeTracker.track
package:build_runner_core/src/generate/build_impl.dart 417:22           _SingleBuild._runForInput.<fn>
package:pool/pool.dart 127:28                                           Pool.withResource
package:build_runner_core/src/generate/build_impl.dart 413:17           _SingleBuild._runForInput
package:build_runner_core/src/generate/build_impl.dart 363:38           _SingleBuild._runBuilder.<fn>
dart:async                                                              Future.wait
package:build_runner_core/src/generate/build_impl.dart 362:36           _SingleBuild._runBuilder
dart:async                                                              _AsyncAwaitCompleter.start
package:build_runner_core/src/generate/build_impl.dart 360:40           _SingleBuild._runBuilder
package:build_runner_core/src/generate/build_impl.dart 309:20           _SingleBuild._runPhases.<fn>.<fn>
dart:async                                                              _completeOnAsyncReturn
package:build_runner_core/src/generate/build_impl.dart                  _SingleBuild._matchingPrimaryInputs
```

**Comment by SamuelDebruyn** [comment]
Looks like this issue has nothing to do with Flutter. There's just a coding error on this [line](https://github.com/Jaguar-dart/jaguar_serializer/blob/42acf0b1f15e313c3bb867ff95a0d78aa8ab95dd/serializer_cli/lib/src/writer/writer.dart#L129) which was [introduced](https://github.com/Jaguar-dart/jaguar_serializer/commit/134f86e9648346afdf4eb383d0e054b03392ba76) in 2.1.1. You can't use previous versions because they're not compatible with Dart 2.

It's checking if `param` is null and if so it's using a getter on `param` which is not possible :)

A workaround can be referencing `jaguar_serializer_cli` from a local path in your pubspec, cloning this repo and replacing that line with `_w.write('null');`

I'm not experienced enough with `build_runner` stuff to do a PR myself to properly fix this.

**Comment by tejainece** [comment]
@SamuelDebruyn Thats for insight into the issue.

I will fix this.

**Comment by tejainece** [comment]
Fixed in commit https://github.com/Jaguar-dart/jaguar_serializer/commit/ea3d986d9025be95c71006583c9903bece3a53f2.

Released as version `jaguar_serializer_cli` `2.2.8`. Please test it and let me if the issue is fixed.

**Comment by SamuelDebruyn** [comment]
@tejainece That fixed it, thanks!
